src/main/webapp/app/tasks/local-task.service.ts

```diff
@@ -32,9 +32,24 @@ export class LocalTaskService implements TaskService {
     return of(null);
   }
 
-  private readTasks(): Task[] {
+  private readTasks(query?:string): Task[] {
```

Comment:
like previously said, having the optional query does complicate things a bit in my opinion